test/core/xds/xds_bootstrap_test.cc

```diff
@@ -55,12 +55,27 @@ namespace grpc_core {
 namespace testing {
 namespace {
 
+MATCHER_P(EqXdsServer, name, "equals XdsServer") {
+  auto* server = static_cast<const GrpcXdsBootstrap::GrpcXdsServer*>(arg);
+  if (!::testing::ExplainMatchResult(::testing::Not(nullptr), server,
```

Comment:
If channel_creds_config is null, we still want to check the URI.  Suggest writing this as:

```
if (!::testing::ExplainMatchResult(::testing::Ne(nullptr), server, result_listener)) {
  return false;
}
bool ok = true;
ok |= ::testing::ExplainMatchResult(name, server->server_uri(), result_listener);
auto creds_config = server->channel_creds_config();
if (!::testing::ExplainMatchResult(::testing::Ne(nullptr), creds_config, result_listener)) {
  return false;
}
ok |= ::testing::ExplainMatchResult("fake", creds_config->type(), result_listener);
return ok;
```